test/core/utils/preload-media.js

```diff
@@ -0,0 +1,76 @@
+/* global Promise */
+
+describe('axe.utils.preloadMedia', function() {
+	'use strict';
+
+	var origFn = axe.utils.preloadMedia;
+	var fixture = document.getElementById('fixture');
+	var fixtureSetup = axe.testUtils.fixtureSetup;
+
+	function rebuildTree() {
+		axe._tree = axe.utils.getFlattenedTree(document);
+	}
+
+	afterEach(function() {
+		axe.utils.preloadMedia = origFn;
+		fixture.innerHTML = '';
+	});
+
+	it('invokes utils.preloadMedia and passes the treeRoot property', function(done) {
+		var isCalled = false;
+		axe.utils.preloadMedia = function(options) {
+			assert.isDefined(options.treeRoot);
+			isCalled = true;
+			return Promise.resolve();
+		};
+
+		rebuildTree();
+
+		axe.utils.preloadMedia({ treeRoot: axe._tree[0] }).then(function() {
+			assert.ok(isCalled);
+			done();
+		});
+	});
+
+	it('returns empty array when there are no media nodes to be preloaded', function(done) {
+		rebuildTree();
+
+		axe.utils.preloadMedia({ treeRoot: axe._tree[0] }).then(function(result) {
+			assert.equal(result.length, 0);
+			done();
+		});
+	});
+
+	it('returns media node (audio) after their metadata has been preloaded', function(done) {
+		fixtureSetup(
+			'<audio src="https://act-rules.github.io/test-assets/moon-audio/moon-speech.mp3" autoplay="true" controls></audio>'
```

Comment:
Put this file into the repo so we're not relying on a remote server that's in no way synched with this repo. Same goes for the rest of the tests below.